src/Mvc/Mvc.NewtonsoftJson/src/NewtonsoftJsonResultExecutor.cs

```diff
@@ -125,6 +125,17 @@ public async Task ExecuteAsync(ActionContext context, JsonResult result)
 
             try
             {
+                var value = result.Value;
+                if (value != null && _asyncEnumerableReaderFactory.TryGetReader(value.GetType(), out var reader))
+                {
+                    Log.BufferingAsyncEnumerable(_logger, value);
+                    value = await reader(value, context.HttpContext.RequestAborted);
```

Comment:
Does this not throw a `OperationCanceledException` if the token is canceled?

Reply:
Not unless the user throws from their IAsyncEnumerable

Reply:
Wouldn't that happen if the async enumerable does `ThrowIfCancelled`?

Reply:
Yes, but do we want to catch an exception from the user or keep the current behavior and let it throw user exceptions? Or third option, catch OCE and check if the token that threw is the same token?

Reply:
Doesn't this apply for the other try-catch too? Like if I registered a JsonConverter that throws OCE instead of ArgNullException?